From 6ed87c61a66225cec6e306f2250c077fb31e98bf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lothar=20Wa=C3=9Fmann?= <LW@KARO-electronics.de>
Date: Tue, 26 Mar 2019 10:40:25 +0100
Subject: [PATCH 12/35] imx8mm: don't clear the reset status upon reading it

The operating system may want to know the last reset cause too. So
don't clear the SRSR register after reading it.
---
 arch/arm/mach-imx/cpu.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/arch/arm/mach-imx/cpu.c b/arch/arm/mach-imx/cpu.c
index e04219f..677866c 100644
--- a/arch/arm/mach-imx/cpu.c
+++ b/arch/arm/mach-imx/cpu.c
@@ -40,18 +40,11 @@ static u32 reset_cause = -1;
 
 static const char *get_reset_cause(void)
 {
-	u32 cause;
 	struct src *src_regs = (struct src *)SRC_BASE_ADDR;
 	struct watchdog_regs *wdog = (struct watchdog_regs *)WDOG1_BASE_ADDR;
+	u32 cause = readl(&src_regs->srsr);
 	u16 wrsr = readw(&wdog->wrsr);
 
-	cause = readl(&src_regs->srsr);
-#ifndef CONFIG_ANDROID_BOOT_IMAGE
-	/* We will read the ssrs states later for android so we don't
-	 * clear the states here.
-	 */
-	writel(cause, &src_regs->srsr);
-#endif
 	reset_cause = cause;
 
 	if (cause == 1)
-- 
2.7.4

